File: section-6-ssl-security-applications/vid-04-php/fetch-record.php
#!/usr/bin/php
<?php

// context
$ctx = stream_context_create(array(
    "ssl" => array(
        "local_cert"        => "/etc/ssl/mongo-client.pem",
        "verify_peer"       => true,
        "cafile"            => "/etc/ssl/mongod.pem",
    ),
));

// login and authenticate as read only user
$m = new MongoClient("mongodb://statsdb_ro:secret@localhost:16687/statsdb", array("ssl" => true), array("context" => $ctx));

// get handle for statsdb and statistics collection
$db = $m->statsdb;
$statistics = $db->statistics;

// retrieve a record and display it
$statistical_record = $statistics->findOne();
var_dump($statistical_record);

?>

File: section-6-ssl-security-applications/vid-05-python/fetch-record.py
#!/usr/bin/env python
from pymongo import MongoClient
import ssl

def main():
    # open connection and authenticate as read only user
    client = MongoClient('mongodb://statsdb_ro:secret@localhost:16687/statsdb',
                         ssl=True,
                         ssl_certfile='/etc/ssl/mongo-client.pem',
                         ssl_ca_certs='/etc/ssl/mongod.pem')

    # get handle for statsdb and statistics collection
    db = client.statsdb
    collection = db.statistics
    
    # retrieve and output one document
    print collection.find_one()

if __name__ == '__main__':
    main()
